client/my-sites/promote-post-i2/components/search-bar/index.tsx

```diff
@@ -204,7 +204,7 @@ export default function SearchBar( props: Props ) {
 		setSearchInput( search );
 		handleSetSearch( {
 			search: search,
-			order: sortOption,
+			order: mode === 'posts' ? sortOption : campaignSortOption,
```

Comment:
this was the issue all along...

when the search is firing we didn't differentiate the 2 different sorting options